standards/core/store-sync.md

```diff
@@ -0,0 +1,52 @@
+---
+title: WAKU-STORE-SYNC
+name: Waku Store Synchronization
+editor: Simon-Pierre Vivier <simvivier@status.im>
+contributors:
+---
+
+## Abstract
+
+This document describe the strategy Waku Store node will employ to stay synchronized.
+The goal being that all store nodes eventually archive the same set of messages.
+
+## Background / Rationale / Motivation
+
+Message propagation in the network is not perfect,
+even with GossipSub mechanisms to detect missed messages.
+Nodes can also go offline for various reason outside our control.
+Store nodes that want to provide a good service must be able to remedy situations like these.
+By having store nodes syncronize with each other through various protocols,
+the set of archived messages network wide will be eventually consistent.
+
+## Theory / Semantics
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “NOT RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+Various protocols and features that help with message consistency are described below.
+
+### Store Resume
+
+This feature allow a node to fill the gap in messages for the period it was last offline.
+At startup, a node use the Store protocol to query a random node for
+the time interval since it was last online.
+Messages returned my the query are then added to the node local archive storage.
+It is RECOMMENDED” to limit the time interval to a maximum of 6 hours.
+
+### Waku Sync
+
+Nodes that stay online can still miss messages.
+[Waku Sync](TODO link to spec) is the protocol used to find those messages by periodically syncing with random nodes.
```

Comment:
Will be good to finish this TODO before merging.

Reply:
Waku Sync, afaics will only be useful to find message _hashes_ and not messages, which is why Store queries are also required.

---

Perhaps not for this PR, but for following versions of the Waku Sync 2.0 protocol it's useful to think of a specification as a heuristic/set of directives that almost form a kind of pseudocode. This becomes easier if you use the RFC keywords for each directive, e.g:

Nodes participating in Store Sync MUST use Waku Sync to find diffs in message hashes with other Store Sync nodes.
The node MUST use Store queries to retrieve the full contents of messages for each message hash missing from its own set.